packages/api/endpoints/granules.js

```diff
@@ -109,7 +152,8 @@ async function put(req, res) {
   const action = body.action;
 
   if (!action) {
-    return res.boom.badRequest('Action is missing');
+    req.body.granuleId = granuleId;
```

Comment:
For updating a granule, req.body.granuleId should already exist and matches with the req.params.granuleName.  We should check that.

Reply:
Agreed the granuleId should already exist. Are you saying we should verify that the body payload matches the url path param (granuleName) and throw an error if it doesn't match?  Because I agree if that's what you're saying.  I was thinking you always wanted to update the granule you PUT to, but if your body doesn't match, you probably don't want to post it.

Reply:
Yeah, we can throw an error if the body.granuleId doesn't match the req.params.granuleName

Reply:
5818774029